sdk/extensions/Microsoft.Extensions.Azure/src/Internal/ClientFactory.cs

```diff
@@ -141,13 +160,37 @@ internal static TokenCredential CreateCredential(IConfiguration configuration, T
                     throw new InvalidOperationException($"Unable to find a certificate with thumbprint '{certificate}'");
                 }
 
-                var credential = new ClientCertificateCredential(tenantId, clientId, certs[0], identityClientOptions);
+                ClientCertificateCredential credential;
+                if (additionallyAllowedTenantsList != null)
+                {
+                    var options = new ClientCertificateCredentialOptions();
+                    foreach (string tenant in additionallyAllowedTenantsList)
+                    {
+                        options.AdditionallyAllowedTenants.Add(tenant);
+                    }
+                    credential = new ClientCertificateCredential(tenantId, clientId, certs[0], options);
+                }
+                else
+                {
+                    credential = new ClientCertificateCredential(tenantId, clientId, certs[0]);
+                }
+
                 store.Close();
 
                 return credential;
             }
 
             // TODO: More logging
+
+            if (additionallyAllowedTenantsList != null)
+            {
+                var options = new DefaultAzureCredentialOptions();
+                foreach (string tenant in additionallyAllowedTenantsList)
+                {
+                    options.AdditionallyAllowedTenants.Add(tenant);
+                }
+                return new DefaultAzureCredential(options);
```

Comment:
Since we are creating the DefaultAzureCredential, should we also pass the other relevant options that can be configured:
- tenantId
- clientId
- managedIdentityResourceId
Any others?

Reply:
If this is a more convenient alternative to providing a full CredentialFactory, I think it's a good change. Although it customers were previously supplying a factory for this scenario, it may break them due to the current behavior of ignoring the factory when an IConfig is present.

Reply:
I think we can reasonably treat the existing behavior where we ignore the config if we don't have all the matching config needed for a credential as a bug.